caffe2/CMakeLists.txt

```diff
@@ -1529,7 +1529,15 @@ if(USE_XPU)
   target_link_libraries(
       torch_xpu PRIVATE ${Caffe2_XPU_DEPENDENCY_LIBS})
 
-  target_link_libraries(torch_xpu PUBLIC torch_cpu_library)
+  # Ensure that torch_cpu is ready before being linked by torch_xpu.
+  add_dependencies(torch_xpu torch_cpu)
+
+  if("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
+    target_link_libraries(torch_xpu PRIVATE
+        "-Wl,--no-as-needed,\"$<TARGET_FILE:torch_cpu>\" -Wl,--as-needed")
+  else()
+    target_link_libraries(torch_xpu PRIVATE "$<TARGET_FILE:torch_cpu>")
```

Comment:
Why not just use 'target_link_libraries(torch_xpu PRIVATE  torch_cpu)' and I think 'add_dependencies(torch_xpu torch_cpu)' is not needed after this change.

Reply:
@cyyever `torch_cpu` contains `caffe2::mkl`, which will result in CPU MKL libraries/options becoming part of XPU link libraries/options. That's also why I want to modify `caffe2::mkl` to private dependency.

Reply:
According to https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#genex:TARGET_FILE, `$<TARGET_FILE:torch_cpu>` will not make `torch_xpu` dependent on `torch_cpu`. Does Clang implicitly ensure that `torch_cpu` is ready before `torch_xpu` build?

Reply:
Thank you for your explanation. The GNU check is not accurate, what you want is to check linker supports '-Wl,--no-as-needed'.